Backend/canvas.js

```diff
@@ -65,10 +65,22 @@ export async function getAssignments(canvas_api_token, course_id) {
       const response = await axios.get(`/courses/${course_id}/assignments`, {
         params: {
           'access_token': canvas_api_token,
-          "per_page": "100"
+          "per_page": "100",
+          "bucket": "unsubmitted",
+          "order_by": "due_at"
         }
       });
-      return response.data;
+
+      var res = [];
+      response.data.map((a) => {
+        var dueAt = new Date(a.due_at)
+
+        if (a.due_at && dueAt > Date.now()) {
+          res.push([a.id, a.name, a.due_at, a.points_possible]);
+        }
+      })
```

Comment:
`map` generally isn't used this way.

```suggestion
      const res = response.data.flatMap((a) => {
        const dueAt = new Date(a.due_at);

        if (a.due_at && dueAt > Date.now()) {
          return [[a.id, a.name, a.due_at, a.points_possible]];
        } else {
          return [];
      })
```

Reply:
Clever solution, replaced.